addons/mail/static/tests/chatter_tests.js

```diff
@@ -520,6 +520,101 @@ QUnit.test('attachmentBox basic rendering', async function (assert) {
     form.destroy();
 });
 
+QUnit.test('attachmentBox drag and drop files [REQUIRE NON-INCOGNITO WINDOW]', async function (assert) {
+    assert.expect(12);
+    this.data.partner.records.push({
+        id: 7,
+        display_name: "attachment_test",
+    });
+
+    testUtils.mock.patch(AttachmentBox, {
+        _callUploadAttachment: function (controllerUrl, formData) {
+            assert.strictEqual(controllerUrl, "/web/binary/upload_attachment",
+                "upload url should match to /web/binary/upload_attachment");
+            assert.strictEqual(formData.get("ufile").name, "text.txt",
+                "uploading file name should be 'text.txt'");
+            this.trigger_up('reload_attachment_box');
+            return Promise.resolve();
```

Comment:
You may omit this and define the function as async, if you really need the async behaviour (which I'm not sure you do)